pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -73,959 +71,762 @@
+            else:
+                rank_zero_warn(
                     "The flag `devices` must be an int with `accelerator='cpu'`,"
-                    f" got `devices={self.devices}` instead."
+                    f" got `devices={self._device_flag}` instead."
                 )
-            self.num_processes = self.devices
-            return True
-        return False
 
-    @property
-    def use_dp(self) -> bool:
-        return self._strategy_type == _StrategyType.DP
+        self._gpus = self._device_flag if not self._gpus else self._gpus
 
-    @property
-    def use_ddp(self) -> bool:
-        return self._strategy_type in (
-            _StrategyType.DDP,
-            _StrategyType.DDP_SPAWN,
-            _StrategyType.DDP_SHARDED,
-            _StrategyType.DDP_SHARDED_SPAWN,
-            _StrategyType.DDP_FULLY_SHARDED,
-            _StrategyType.DEEPSPEED,
-            _StrategyType.TPU_SPAWN,
-        )
+    def _choose_and_init_cluster_environment(self):
+        self.cluster_environment = LightningEnvironment()
+        if isinstance(self._cluster_environment, ClusterEnvironment):
+            self.cluster_environment = self._cluster_environment
+        elif self._is_slurm_managing_tasks():
+            rank_zero_info("Multiprocessing is handled by SLURM.")
+            self.cluster_environment = SLURMEnvironment()
+        else:
+            for env_type in (TorchElasticEnvironment, KubeflowEnvironment, LSFEnvironment):
+                if env_type.detect():
+                    self.cluster_environment = env_type()
 
-    @property
-    def use_ddp2(self) -> bool:
-        return self._strategy_type == _StrategyType.DDP2
+    def _is_slurm_managing_tasks(self):
+        """used by choosing cluster enviroment."""
+        if not SLURMEnvironment.detect() or SLURMEnvironment.job_name() == "bash":
+            return False
 
-    @property
-    def use_horovod(self) -> bool:
-        return self._strategy_type == _StrategyType.HOROVOD
+        total_requested_devices = len(self._parallel_devices) * self._num_nodes_flag
+        num_slurm_tasks = int(os.environ["SLURM_NTASKS"], 0)
+        return num_slurm_tasks == total_requested_devices
 
-    @property
-    def use_deepspeed(self) -> bool:
-        return self._strategy_type == _StrategyType.DEEPSPEED
+    def _choose_strategy(self):
+        if self._accelerator_flag == "ipu":
+            self._strategy_flag = "ipu"
+        elif self._accelerator_flag == "tpu":
+            if self._parallel_devices and len(self._parallel_devices) > 1:
+                self._strategy_flag = "tpu_spawn"
+            else:
+                # TODO lazy initialized device, then here could be self._strategy_flag = "single_tpu_device"
+                self._strategy_flag = SingleTPUStrategy(device=self._parallel_devices[0])
+        elif _HOROVOD_AVAILABLE and ("OMPI_COMM_WORLD_RANK" in os.environ or "HOROVOD_RANK" in os.environ):
```

Comment:
@awaelchli does it make sense to have a horovod cluster environment?

Reply:
let's discuss this in a separate issue?